### PR TITLE
Move instructions about how to run the storage operator

### LINKED ISSUE #1536: Document how to run stuff locally
**Component**:

documentation

**Why this is needed**:

There are several components that we can run locally, and:
- sometime it's documented in our documentation (setup a cluster with Vagrant)
- sometimes it's documented in a README (running the storage operator)
- sometimes it's not documented at all (running the UI locally).

**What should be done**:

Document everything in a single place (our documentation) under a dedicated section of the developers documentation.

**Implementation proposal** (strongly recommended):

- Create a new section "How to run components locally" in the "Developer Guide"
- Move the documentation on `vagrant_up` there
- Add a section for the UI
- Convert the `README.md` of the storage operator into ReST and move it here (from 2.4 only)

**Test plan**:

Following the documentation, one should be able to spawn a cluster/the UI/the storage operator locally.

### PR DESCRIPTION
**Pull request by slaperche-scality**
**Component**:

documentation

**Context**: 

Avoid scattering the documentation all over the repository, centralize it into the `doc` subdirectory.

**Summary**:

Move the content from `storage-operator/README.md` into the Developer Guide.

**Acceptance criteria**: 

The documentation now has a page on how to run the operator.

---

Closes: #1536

### CODE REVIEW

File: docs/developer/running/storage_operator.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
+Running the storage operator locally
+====================================
+
+Requirements
+------------
+
+- `Go <https://golang.org/>`_ (1.12 or higher) and
+  `operator-sdk <https://github.com/operator-framework/operator-sdk>`_ (0.9 or
+  higher): to build the Kubernetes Operators
+- `Mercurial <https://www.mercurial-scm.org/>`_: some Go dependencies are
```
Comment (jbertran) [inline docs/developer/running/storage_operator.rst:10]: For comprehensiveness: `git` is a dependency too ^^

File: docs/developer/running/storage_operator.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  ``cd storage-operator; go mod download``
+
+Procedure
+---------
+
+1. Copy the ``/etc/kubernetes/admin.conf`` from the bootstrap node of your
+   cluster onto your local machine
+
+2. Delete the already running storage operator, if any, with
+   ``kubectl --kubeconfig /etc/kubernetes/admin.conf -n kube-system
```
Comment (jbertran) [inline docs/developer/running/storage_operator.rst:27]: This way we explicitly stay on the local machine as soon as we've finished grabbing the admin.conf from the remote bootstrap node.

```suggestion
   ``kubectl --kubeconfig <PATH_TO_LOCAL_ADMIN_CONF> -n kube-system
```

File: docs/developer/running/storage_operator.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+   delete deployment storage-operator``
+
+3. Get the address of the Salt API server with
+   ``kubectl --kubeconfig /etc/kubernetes/admin.conf -n kube-system
```
Comment (jbertran) [inline docs/developer/running/storage_operator.rst:31]: Ditto here.

File: docs/developer/running/storage_operator.rst (hunk 1 above)
Comment (slaperche-scality) [reply docs/developer/running/storage_operator.rst:10]: Yeah, but if you cloned the repo it's expected that you have `Git` ^^

If we go that way, a computer is required as well :stuck_out_tongue:

File: docs/developer/running/storage_operator.rst (hunk 2 above)
Comment (slaperche-scality) [reply docs/developer/running/storage_operator.rst:27]: Yeah but not everyone have `kubectl` installed locally, so I prefer to document the way from the bootstrap node.

File: docs/developer/running/storage_operator.rst (hunk 3 above)
Comment (slaperche-scality) [reply docs/developer/running/storage_operator.rst:31]: Ditto ^^

**Review by jbertran — APPROVED** [review]
Fair enough :)

**Pull request merged**